config/config_test.go

```diff
@@ -0,0 +1,159 @@
+package config
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func emptyConfig() *Config {
+	return &Config{ResourceType{FilterRule{}, FilterRule{}}}
+}
+
+func TestConfig_Garbage(t *testing.T) {
+	configFilePath := "./mocks/garbage.yaml"
+	configObj, err := GetConfig(configFilePath)
+
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	if !reflect.DeepEqual(configObj, emptyConfig()) {
+		assert.Fail(t, "Config should be empty, %+v\n", configObj)
+	}
+
+	return
+}
+
+func TestConfig_Malformed(t *testing.T) {
+	configFilePath := "./mocks/malformed.yaml"
+	_, err := GetConfig(configFilePath)
+
+	// Expect malformed to throw a yaml TypeError
+	if err != nil {
+		require.Error(t, err, "Received expected error")
+	}
+
+	require.Error(t, err, "Expected error")
+	return
+}
+
+func TestConfig_Empty(t *testing.T) {
+	configFilePath := "./mocks/empty.yaml"
+	configObj, err := GetConfig(configFilePath)
+
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	if !reflect.DeepEqual(configObj, emptyConfig()) {
+		assert.Fail(t, "Config should be empty, %+v\n", configObj)
+	}
+
+	return
+}
+
+func TestConfigS3_Empty(t *testing.T) {
+	configFilePath := "./mocks/s3_empty.yaml"
+	configObj, err := GetConfig(configFilePath)
+
+	if err != nil {
+		require.NoError(t, err)
+	}
+
+	if !reflect.DeepEqual(configObj, emptyConfig()) {
+		assert.Fail(t, "Config should be empty, %+v\n", configObj.S3)
+	}
+
+	return
+}
+
+func TestConfigS3_EmptyFilters(t *testing.T) {
+	configFilePath := "./mocks/s3_empty_filters.yaml"
+	configObj, err := GetConfig(configFilePath)
+
+	if err != nil {
```

Comment:
NIT: don't need this if statement. Can just call `require.NoError` directly.

Reply:
- [ ] TODO